src/ptp/ptpd2/protocol.c

```diff
@@ -1129,6 +1134,16 @@ processPortMessage(RunTimeOpts *rtOpts, PtpClock *ptpClock,
 	isFromSelf = (ptpClock->portIdentity.portNumber == ptpInterface->msgTmpHeader.sourcePortIdentity.portNumber
 		      && !memcmp(ptpInterface->msgTmpHeader.sourcePortIdentity.clockIdentity, ptpClock->portIdentity.clockIdentity, CLOCK_IDENTITY_LENGTH));
 
+	if (isFromSelf) {
+		struct sfptpd_ts_ticket ticket;
+		struct sfptpd_ts_user user;
+
+		ticket = netMatchPacketToTsCache(&ptpInterface->ts_cache,
+						 &user,
+						 ptpInterface->msgIbuf, length);
+		processTxTimestamp(ptpInterface, user, ticket, *timestamp);
+	}
```

Comment:
Probably we should return at this point as we no longer need to process the packet further if this is a looped back timestamp.